src/invidious/videos.cr

```diff
@@ -326,13 +326,18 @@ end
 def fetch_video(id, region)
   info = extract_video_info(video_id: id)
 
-  if reason = info["reason"]?
+  if info["reason"]? && info["subreason"]?
+    reason = info["reason"].as_s
+    puts info
+    if info.dig?("subreason").nil?
+      subreason = info["subreason"].as_s
+    else
+      subreason = "No additional reason"
+    end
```

Comment:
This bloc of code can be summarized to the following, as you're already checking `info["subreason"]?` falsiness in the parent `if`
```suggestion
    subreason = info["subreason"].as_s
```